File: 1443/1443.cs
public class Solution {
    public int MinimumTimeToCollectAllApplesI(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MinimumTimeToCollectAllApplesI(root.left), MinimumTimeToCollectAllApplesI(root.right));
    }
}

File: 1443/1443.py
class Solution:
    def minimum_time_to_collect_all_apples_in_a_(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.minimum_time_to_collect_all_apples_in_a_(root.left), self.minimum_time_to_collect_all_apples_in_a_(root.right))